### PR TITLE
Chat input area component

### PR DESCRIPTION
**Pull request by burieberry**
1- Kept the hover and focus states to remain consistent with rest of boxel inputs and also for better accessibility
2- ~Pressing "enter" submits the text and "shift enter" creates a line break.~ Just noticed that in one of the designs it says "cmd enter" should submit the text, I will update. Updated: `cmd+Enter` or `ctrl+Enter` submits the message. `enter` creates line breaks.
3- The trick I used to expand textarea only works when there are line breaks. the only other way to do this without line breaks requires not using a `textarea` but casting a generic element as a contenteditable textbox, which is not great for accessibility and does not come with all the bindings.

From the ticket [CS 6424]:
- [x] Initial state of text field accommodates 2 lines of text
- [x] Text field expands VERTICALLY to accommodate up to 5 lines
- [x] Scroll for text that exceeds 5 lines

empty

hover state:
hover

focus state:
focus

no scroll up to 5 lines:
5-lines

scrolls over 5 lines:
scroll

### CODE REVIEW

File: packages/host/app/components/ai-assistant/chat-input/index.gts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+interface Signature {
+  Element: HTMLDivElement;
+  Args: {
+    value: string;
+    onInput: (val: string) => void;
+    onSend: (val: string) => void;
+  };
+}
+
+// @keyResponder
```
Comment (backspace) [inline packages/host/app/components/ai-assistant/chat-input/index.gts]: Could this be removed?

File: packages/host/app/components/ai-assistant/chat-input/index.gts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      .send-button {
+        --icon-color: var(--boxel-highlight);
+        height: 30px;
+        padding-bottom: var(--boxel-sp-xs);
+      }
+      .send-button:hover:not(:disabled) {
+        --icon-color: var(--boxel-highlight-hover);
+      }
+      .send-button > :deep(svg) {
+        margin: 0;
```
Comment (backspace) [inline packages/host/app/components/ai-assistant/chat-input/index.gts]: `IconButton` [overrides this](https://github.com/cardstack/boxel/blob/121186e11093fe057c15ec533f93ab99146abee8/packages/boxel-ui/addon/src/components/icon-button/index.gts#L46) so it seems unnecessary?

Host Components 2024-01-09 13-01-41

File: packages/host/types/ember-keyboard/modifiers/on-key.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    Positional: [keyCombo: string, callback?: (ev: KeyboardEvent) => void];
+    Named: {
+      event?: string;
+      activated?: boolean;
+      priority?: number;
+    };
+  };
+}
+
+export default class OnKeyModifier extends Modifier<Signature> {}
```
Comment (backspace) [inline packages/host/types/ember-keyboard/modifiers/on-key.d.ts:15]: If we produce enough of these maybe we can just [do this](https://github.com/adopted-ember-addons/ember-keyboard/issues/464#issuecomment-1709156176) 😆

**Review by backspace — APPROVED** [review]
This looks good, I haven’t seen that derived `height` before, works well! 🤩

File: packages/host/app/components/ai-assistant/chat-input/index.gts (hunk 1 above)
Comment (burieberry) [reply packages/host/app/components/ai-assistant/chat-input/index.gts]: good catch, removed

File: packages/host/app/components/ai-assistant/chat-input/index.gts (hunk 2 above)
Comment (burieberry) [reply packages/host/app/components/ai-assistant/chat-input/index.gts]: yes, meant to remove that as well

**Pull request merged**